templates/pyproject.toml

```diff
@@ -0,0 +1,205 @@
+    #    the text, or leave a blank line between the last text and the
+    # quotes)
+    "GL02",  # Permit a blank line before docstring closing quotes.
+
+    #    -> Double line break found; please use only one blank line to
+    #    separate sections or paragraphs, and do not leave blank lines
+    #    at the end of docstrings
+    "GL03",  # Ignoring.
+
+    # -> See Also section not found
+    "SA01",  # Not all docstrings require a "See Also" section.
+
+    # -> No extended summary found
+    "ES01",  # Not all docstrings require an "Extended Summary" section.
+
+    # -> No examples section found
+    "EX01",  # Not all docstrings require an "Examples" section.
+
+    # -> No Yields section found
+    "YD01",  # Not all docstrings require a "Yields" section.
+
+]
+exclude = [
+    '\.__eq__$',
+    '\.__ne__$',
+    '\.__repr__$',
+]
+
+[tool.pytest.ini_options]
+# This list varies by repository
+addopts = ["-ra", "-v", "--strict-config", "--strict-markers", "--doctest-modules"]
+testpaths = "<PATH_TO_MAIN_CODE>" # e.g "cf_units" or "lib/iris"
+
+[tool.repo-review]
+# A list of the currently failing repo-review checks, to be fixed later, different for every repository.
+ignore = []
+
+[tool.ruff]
+line-length = 88
+
+[tool.ruff.format]
+preview = false
+
+[tool.ruff.lint]
+# We have agreed that we will permanently ignore the ruff rules in this ignore list.
+ignore = [
+    # NOTE: Non-permanent exclusions should be added to the ".ruff.toml" file.
+
+    # flake8-commas (COM)
+    # https://docs.astral.sh/ruff/rules/#flake8-commas-com
+    "COM812",  # Trailing comma missing.
+    "COM819",  # Trailing comma prohibited.
+
+    # flake8-implicit-str-concat (ISC)
+    # https://docs.astral.sh/ruff/rules/single-line-implicit-string-concatenation/
+    # NOTE: This rule may cause conflicts when used with "ruff format".
+    "ISC001",  # Implicitly concatenate string literals on one line.
+]
+preview = false
+select = [
+    "ALL",
+    # Note: the above "all" disables conflicting rules, if you want that
+    # rule it needs to be explicitly enabled below:
+    "D212",  # Multi-line docstring summary should start at the first line
+]
+
+[tool.ruff.lint.isort]
+force-sort-within-sections = true
+known-first-party = ["<PACKAGE_NAME>"]  # e.g "iris" or "cf_units"
+
+[tool.ruff.lint.mccabe]
+max-complexity = 22
+
+[tool.ruff.lint.per-file-ignores]
+# All test scripts
+"cf_units/tests/*.py" = [
+    # https://docs.astral.sh/ruff/rules/undocumented-public-module/
+    "D104",  # Missing docstring in public module
+    "D106",  # Missing docstring in public nested class
+    "D205",  # 1 blank line required between summary line and description
+    "D401",  # 1 First line of docstring should be in imperative mood
+    "SLOT000",  # Subclasses of `str` should define `__slots__`
+    "N999",  # Invalid module name
+]
+"setup.py" = [
+    "FBT003",  # Boolean positional value in function call
+    "ICN001",  # `numpy` should be imported as `np`
+]
```

Comment:
This is a deviation from the standardised settings in [the wiki](https://github.com/SciTools/.github/wiki/Linting#ruff-2); unfortunately I do not remember the history. Let me know if you're certain this is correct